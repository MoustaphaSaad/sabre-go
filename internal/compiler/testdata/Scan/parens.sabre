// Parentheses and brackets
array := [10]int32{1, 2, 3}
matrix := mat4x4{{1, 0, 0, 0}, {0, 1, 0, 0}}
function_call := sqrt(x * x + y * y)
grouping := (a + b) * (c - d)

// Braces
block := {
    nested := {
        deeply := {
            value
        }
    }
}
