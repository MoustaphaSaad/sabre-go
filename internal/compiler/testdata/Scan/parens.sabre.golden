COMMENT         "// Parentheses and brackets"    1:1       1:28   [0-27]
;               "\n"                    1:28      2:1    [27-28]
IDENTIFIER      "array"                 2:1       2:6    [28-33]
:=              ":="                    2:7       2:9    [34-36]
[               "["                     2:10      2:11   [37-38]
LITERAL_INT     "10"                    2:11      2:13   [38-40]
]               "]"                     2:13      2:14   [40-41]
IDENTIFIER      "int32"                 2:14      2:19   [41-46]
{               "{"                     2:19      2:20   [46-47]
LITERAL_INT     "1"                     2:20      2:21   [47-48]
,               ","                     2:21      2:22   [48-49]
LITERAL_INT     "2"                     2:23      2:24   [50-51]
,               ","                     2:24      2:25   [51-52]
LITERAL_INT     "3"                     2:26      2:27   [53-54]
}               "}"                     2:27      2:28   [54-55]
;               "\n"                    2:28      3:1    [55-56]
IDENTIFIER      "matrix"                3:1       3:7    [56-62]
:=              ":="                    3:8       3:10   [63-65]
IDENTIFIER      "mat4x4"                3:11      3:17   [66-72]
{               "{"                     3:17      3:18   [72-73]
{               "{"                     3:18      3:19   [73-74]
LITERAL_INT     "1"                     3:19      3:20   [74-75]
,               ","                     3:20      3:21   [75-76]
LITERAL_INT     "0"                     3:22      3:23   [77-78]
,               ","                     3:23      3:24   [78-79]
LITERAL_INT     "0"                     3:25      3:26   [80-81]
,               ","                     3:26      3:27   [81-82]
LITERAL_INT     "0"                     3:28      3:29   [83-84]
}               "}"                     3:29      3:30   [84-85]
,               ","                     3:30      3:31   [85-86]
{               "{"                     3:32      3:33   [87-88]
LITERAL_INT     "0"                     3:33      3:34   [88-89]
,               ","                     3:34      3:35   [89-90]
LITERAL_INT     "1"                     3:36      3:37   [91-92]
,               ","                     3:37      3:38   [92-93]
LITERAL_INT     "0"                     3:39      3:40   [94-95]
,               ","                     3:40      3:41   [95-96]
LITERAL_INT     "0"                     3:42      3:43   [97-98]
}               "}"                     3:43      3:44   [98-99]
}               "}"                     3:44      3:45   [99-100]
;               "\n"                    3:45      4:1    [100-101]
IDENTIFIER      "function_call"         4:1       4:14   [101-114]
:=              ":="                    4:15      4:17   [115-117]
IDENTIFIER      "sqrt"                  4:18      4:22   [118-122]
(               "("                     4:22      4:23   [122-123]
IDENTIFIER      "x"                     4:23      4:24   [123-124]
*               "*"                     4:25      4:26   [125-126]
IDENTIFIER      "x"                     4:27      4:28   [127-128]
+               "+"                     4:29      4:30   [129-130]
IDENTIFIER      "y"                     4:31      4:32   [131-132]
*               "*"                     4:33      4:34   [133-134]
IDENTIFIER      "y"                     4:35      4:36   [135-136]
)               ")"                     4:36      4:37   [136-137]
;               "\n"                    4:37      5:1    [137-138]
IDENTIFIER      "grouping"              5:1       5:9    [138-146]
:=              ":="                    5:10      5:12   [147-149]
(               "("                     5:13      5:14   [150-151]
IDENTIFIER      "a"                     5:14      5:15   [151-152]
+               "+"                     5:16      5:17   [153-154]
IDENTIFIER      "b"                     5:18      5:19   [155-156]
)               ")"                     5:19      5:20   [156-157]
*               "*"                     5:21      5:22   [158-159]
(               "("                     5:23      5:24   [160-161]
IDENTIFIER      "c"                     5:24      5:25   [161-162]
-               "-"                     5:26      5:27   [163-164]
IDENTIFIER      "d"                     5:28      5:29   [165-166]
)               ")"                     5:29      5:30   [166-167]
;               "\n"                    5:30      6:1    [167-168]
COMMENT         "// Braces"             7:1       7:10   [169-178]
;               "\n"                    7:10      8:1    [178-179]
IDENTIFIER      "block"                 8:1       8:6    [179-184]
:=              ":="                    8:7       8:9    [185-187]
{               "{"                     8:10      8:11   [188-189]
IDENTIFIER      "nested"                9:5       9:11   [194-200]
:=              ":="                    9:12      9:14   [201-203]
{               "{"                     9:15      9:16   [204-205]
IDENTIFIER      "deeply"               10:9      10:15   [214-220]
:=              ":="                   10:16     10:18   [221-223]
{               "{"                    10:19     10:20   [224-225]
IDENTIFIER      "value"                11:13     11:18   [238-243]
;               "\n"                   11:18     12:1    [243-244]
}               "}"                    12:9      12:10   [252-253]
;               "\n"                   12:10     13:1    [253-254]
}               "}"                    13:5      13:6    [258-259]
;               "\n"                   13:6      14:1    [259-260]
}               "}"                    14:1      14:2    [260-261]
;               "\n"                   14:2      15:1    [261-262]
EOF             ""                     15:1      15:1    [262-262]
